librosa/core/time_frequency.py

```diff
@@ -1682,3 +1713,222 @@ def samples_like(X, hop_length=512, n_fft=None, axis=-1):
     else:
         frames = np.arange(X.shape[axis])
     return frames_to_samples(frames, hop_length=hop_length, n_fft=n_fft)
+
+
+@cache(level=10)
+def key_to_notes(key, unicode=True): 
+    '''Construct the names for 12 chromatic notes for a given key.
+
+    This function exists to resolve enharmonic equivalences between different
+    spellings for the same pitch (e.g. C♯ vs D♭), and is primarily useful when producing
+    human-readable outputs (e.g. plotting) for pitch content.
+
+    Note names are decided by the following rules:
+
+    1. If the tonic of the key has an accidental (sharp or flat), that accidental will be
+       used consistently for all notes.
+
+    2. If the tonic does not have an accidental, accidentals will be inferred to minimize
+       the total number used for diatonic scale degrees.
+
+    3. If there is a tie (e.g., in the case of C:maj vs A:min), sharps will be preferred
+       for major keys, and flats will be preferred for minor keys.
+
```

Comment:
we must say somewhere that the first element of the returned list is enharmonically equivalent to a C, whatever be the key. (users might assume that this element is the root)

Reply:
The `returns` section states this, but I think the examples make the point more clearly.  Do we really need more?

Reply:
Yes, i would believe so. Users might not read that far.
Here's an attempt:

```Lists all 12 note names in the chromatic scale, as spelled according to a given key (major or minor)```

Hopefully the definite article ("_the_ chromatic scale", as opposed to "12 chromatic notes") nudges users into the fact that they are all different spellings of the same MIDI numbers.